server/app/filters/ValidAccountFilter.java

```diff
@@ -47,7 +47,7 @@ public EssentialAction apply(EssentialAction next) {
    * infinite redirect.
    */
   private boolean allowedEndpoint(Http.RequestHeader requestHeader) {
-    return NonUserRoutePrefixes.anyMatch(requestHeader) || isLogoutRequest(requestHeader.uri());
+    return NonUserRoutes.anyMatch(requestHeader) || isLogoutRequest(requestHeader.uri());
```

Comment:
Oh wait this is the other place it's used, not SettingsFilter. I think we wouldn't want `/programs` to be included here since this is only reached if `profile.isPresent` returns true.